v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/ddl/DdlTest.java

```diff
@@ -322,6 +325,144 @@ public void testInterleavingPG() {
     assertNotNull(ddl.hashCode());
   }
 
+  /**
+   * Generates a Spanner DDL object representing a schema from a directed acyclic graph (DAG) of
+   * table dependencies.
+   *
+   * <p>This method creates a simplified DDL with minimal schema information. Each table has only an
+   * "id" column as the primary key. Foreign key relationships are established based on the provided
+   * dependencies.
+   *
+   * @param tableNames A list of table names in the schema.
+   * @param dependencies A list of foreign key dependencies, where each dependency is represented as
+   *     a list of two strings: the child table name and the parent table name, where child
+   *     references the parent.
+   */
+  private Ddl generateDdlFromDAG(List<String> tableNames, List<List<String>> dependencies) {
```

Comment:
Can you add a test for the cyclic dependency also ?

Reply:
Its already there at the end